cmp/tab/TabContainerModel.ts

```diff
@@ -20,85 +87,50 @@ import {TabModel} from './TabModel';
+    switcher: TabSwitcherProps;
+    track: boolean;
+    renderMode: RenderMode;
+    refreshMode: RefreshMode;
+    emptyText: ReactNode;
+    provider: PersistenceProvider;
 
-    /** @member {(string|ReactNode)} */
-    emptyText;
-
-    /** @member {RefreshContextModel} */
     @managed
-    refreshContextModel;
+    refreshContextModel: RefreshContextModel;
 
-    _lastActiveTabId;
+    private _lastActiveTabId;
 
-    /**
-     * @param {Object} c - TabContainerModel configuration.
-     * @param {Object[]} [c.tabs] - configs for TabModels to be displayed.
-     * @param {?string} [c.defaultTabId] - ID of Tab to be shown initially if routing does not
-     *      specify otherwise. If not set, will default to first tab in the provided collection.
-     * @param {?string} [c.route] - base route name for this container. If set, this container will
-     *      be route-enabled, with the route for each tab being "[route]/[tab.id]".
-     *      Cannot be used with `persistWith`.
-     * @param {(boolean|Object)} [c.switcher] - indicates whether to include a default switcher
-     *      docked within this component. Specify as a boolean or an object containing props for a
-     *      TabSwitcher component. Set to false to not include a switcher. Defaults to true.
-     * @param {boolean} [c.track] - True to enable activity tracking of tab views (default false).
-     *      Viewing of each tab will be tracked with the `oncePerSession` flag, to avoid duplication.
-     * @param {RenderMode} [c.renderMode] - strategy for rendering child tabs. Can be set
-     *      per-tab via `TabModel.renderMode`. See enum for description of supported modes.
-     * @param {RefreshMode} [c.refreshMode] - strategy for refreshing child tabs. Can be set
-     *      per-tab via `TabModel.refreshMode`. See enum for description of supported modes.
-     * @param {PersistOptions} [c.persistWith] - options governing persistence.
-     *      Cannot be used with `route`.
-     * @param {(string|ReactNode)} [c.emptyText] - placeholder to display if no tabs are provided
-     *      or all tabs have been removed via their `omit` config.
-     */
     constructor({
         tabs = [],
         defaultTabId = null,
         route = null,
         switcher = true,
         track = false,
-        renderMode = RenderMode.LAZY,
-        refreshMode = RefreshMode.ON_SHOW_LAZY,
+        renderMode = 'lazy',
+        refreshMode = 'onShowLazy',
         persistWith,
         emptyText = 'No tabs to display.',
         xhImpl = false
-    }) {
+    }: TabContainerConfig) {
         super();
         makeObservable(this);
         this.xhImpl = xhImpl;
 
-        // Create default switcher props
-        if (switcher === true) {
-            switcher = {orientation: XH.isMobileApp ? 'bottom' : 'top'};
-        }
-
-        throwIf(switcher && !['top', 'bottom', 'left', 'right'].includes(switcher.orientation), 'Unsupported value for switcher orientation.');
```

Comment:
Not sure we should lose this throw, until apps switch over to TS at least.

Reply:
yeah, that's fair -- but saw that this was already an option on the switcher itself (which I left).  So felt like that two of these checks was overkill.